modules/account-lib/src/coin/hbar/utils.ts

```diff
@@ -65,17 +66,17 @@ export function isValidPublicKey(key: string): boolean {
  * @returns {string} - the hex value
  */
 export function toHex(buffer: Buffer | Uint8Array): string {
-  return Buffer.from(buffer).toString('hex');
+  return hex.encode(buffer, true);
 }
 
 /**
  * Returns a Uint8Array of the given hex string
  *
- * @param {string} hex - the hex string to be converted
+ * @param {string} str - the hex string to be converted
  * @returns {string} - the Uint8Array value
  */
-export function toUint8Array(hex: string): Uint8Array {
-  return Uint8Array.from(Buffer.from(hex, 'hex'));
+export function toUint8Array(str: string): Uint8Array {
+  return hex.decode(str);
```

Comment:
what happens if we don't pass a hex string to this library? does it throw an error? or does it just run through and give garbage?

Reply:
Yup: https://github.com/StableLib/stablelib/blob/master/packages/hex/hex.ts

Reply:
god this is terribly coded. i'll try to figure out this bit shifting later. this looks like perl developers invaded JS

Reply:
also we'll need to figure out why `buffer.from(x, 'hex')` doesn't work for this lib